### ISSUE TITLE
Presets are not expanded with `useBaseBranchConfig: merge`

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#22956]
### Discussed in https://github.com/renovatebot/renovate/discussions/22919

<div type='discussions-op-text'>

<sup>Originally posted by **anomiex** June 22, 2023</sup>
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

When setting `useBaseBranchConfig` to `merge`, the configuration merged from the base branch does not have `extends` processed. This results in rules being misinterpreted.

In the minimal reproduction repo at https://github.com/anomiex/renovate-test-2/, for example, the package rule
```
        {
            extends: [ 'packages:jsUnitTest' ],
            groupName: 'JS unit testing packages',
        },
```
does not have the preset expanded, and so winds up matching _every_ package instead of only the packages specified by the packages:jsUnitTest preset. So we get strange PRs like https://github.com/anomiex/renovate-test-2/pull/16 and https://github.com/anomiex/renovate-test-2/pull/15 instead of expected PRs like https://github.com/anomiex/renovate-test-2/pull/12.

### Have you created a minimal reproduction repository?

Updated minimum reproduction repo by @rarkins: https://github.com/renovate-reproductions/22919

**Comment by rarkins** [comment]
Fixed by #22958

**Comment by HonkingGoose** [comment]
And here's the version of Renovate that has the fix: [^1]

> 🎉 This PR is included in version 35.141.2 🎉
> 
> The release is available on:
> 
>     * [GitHub release](https://github.com/renovatebot/renovate/releases/tag/35.141.2)
> 
>     * `35.141.2`
> 
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

[^1]: https://github.com/renovatebot/renovate/pull/22958#issuecomment-1605331249
